driftbase/api/friendships.py

```diff
@@ -122,7 +122,7 @@ def post(self, args, player_id):
             if friendship.status == "deleted":
                 friendship.status = "active"
             else:
-                return "{}", http_client.OK
+                abort(http_client.CONFLICT, message="You are already friends with this player!")
```

Comment:
This is an API change and we'd need to see how old clients deal with this situation

Reply:
Let's assume I want to do just that, i.e. change an API; 
*  I need a list of all the tenants, presumably from the live tier config
*  I find the repos for the clients and read the code; lets assume some of them look fine and should be ok with the change, but some would need minor updates
* In either case, I'd want to test a client against a dev tier (or local) backend to see if all is good with the API change (with or without client modifications).
* That means I'd need the full build environment for all tenants for local testing and to know how to direct them to the correct testing tier (is driftUrl=XXX in .ini the standard way and works everywhere)?
* I'd also need to know the deployment pipeline to publish updates to live for all the other tenants.

Sounds like a lot of time consuming busywork.

Reply:
Since you can't realistically check all clients, you might not even have access, the only proper way is to not break APIs, but to version, one way or another.

We know we want versioning, and I would prefer not having to operate multiple versions of every deployable forsaking multi-tenancy, as that suddenly requires versioning the deployment infrastructure instead, and you're basically just moving complexity from the API management to infrastructure management and ops.

So, what we should do, is to figure out API versioning. And until then, we need to simply avoid breaking APIs.